src/utils/CdnUtils.ts

```diff
@@ -0,0 +1,55 @@
+import axios from "axios";
+import R2Error from "../model/errors/R2Error";
+
+const CDNS = [
+    "gcdn.thunderstore.io",
+    "hcdn-1.hcdn.thunderstore.io"
+]
+const TEST_FILE = "healthz";
+
+const canReachCdn = async (domain: string) => {
+    const url = `https://${domain}/${TEST_FILE}`;
+    const headers = {
+        "Cache-Control": "no-cache",
+        "Pragma": "no-cache",
+        "Expires": "0",
+    };
+    const params = {"disableCache": new Date().getTime()}
+    let res;
+
+    try {
+        res = await axios.get(url, {headers, params});
```

Comment:
We probably want to add a timeout of 5s or so here, otherwise the initial app startup could freeze for who knows how long as it's pending for CDN checks (given that CDN checks are blocking the app startup currently).

Axios doesn't have a timeout at all by default if I remember correctly.

Reply:
Now implemented with [helper utils](https://github.com/ebkr/r2modmanPlus/pull/1130/commits/6f9bb959392a53500fbda3a16e92dcb8255569bc)